### DISCUSSION TITLE
PIXIJS Graphic  How to realize the effect of hole digging

### DISCUSSION THREAD

**Discussion by pcbren1** [pixijs/pixijs#7567] [Q&A] [answered]
PIXIJS Graphic  How to realize the effect of hole digging

![122](https://user-images.githubusercontent.com/86771584/124097332-0e983780-da8e-11eb-9e50-8fd76841f571.jpg)

**Comment by ivanpopelyshev** [comment, marked as answer]
Simple. Just try to do it in canvas2d first. Then you'll notice its not possible to do through graphics/paths, because its not representable by both winding rules. You have to use some layering and blendModes.

put it all in container with AlphaFilter on it, specify blendMode `BLEND_MODES.ERASE` for third element.

**Reply by pcbren1** [reply]
It's my first time to ask a question in GitHub. Thank you very much for your quick answer.

I'm using Pixi to make a drawing tool, and I have some problems in the process. I still have a few unsolved problems and I'm going to raise them together. Thank you very much again

**Reply by eXponenta** [reply]
Tess2 tesselation backend can solve this easy.

Tess2 wasm is very powerfull:
https://github.com/eXponenta/emscripten-libtess2

**Comment by ivanpopelyshev** [comment]
> I'm using Pixi to make a drawing tool

then you have to understand how layering (RenderTexture!) and blendMode work :) ERASE was added specifically for drawing apps. Also if you want to make brushes with Graphics, I recommend my new plugin https://github.com/pixijs/graphics-smooth , which allows to draw shapes without enabling "antialias"
